src/golang/lib/collections/artifact/types.go

```diff
@@ -15,3 +15,11 @@ const (
 	Image     Type = "image"
 	Picklable Type = "picklable"
 )
+
+// `IsCompact` indicates if the value is 'small' enough to pass around in-memory.
```

Comment:
Continuing the discussion in #470 , for now I decided to determine if value is 'small' only on type. The motivation is that if we need to assume the content with certain type could be large, we shouldn't even assume the content would fit in memory and thus there's no point to read it in memory and determine based on size of value.

In future, we could expose more `storage` API to return the content size, or persist size in DB, to optimize the behavior here.